SDL/TrackCandidate.h

```diff
@@ -415,10 +421,19 @@ namespace SDL
 
                 unsigned int trackCandidateIdx = alpaka::atomicOp<alpaka::AtomicAdd>(acc, trackCandidatesInGPU.nTrackCandidates, 1);
                 alpaka::atomicOp<alpaka::AtomicAdd>(acc, trackCandidatesInGPU.nTrackCandidatespT3, 1);
-
-                float radius = 0.5f * (__H2F(pixelTripletsInGPU.pixelRadius[pixelTripletIndex]) + __H2F(pixelTripletsInGPU.tripletRadius[pixelTripletIndex]));
-                unsigned int pT3PixelIndex =  pixelTripletsInGPU.pixelSegmentIndices[pixelTripletIndex];
-                addTrackCandidateToMemory(trackCandidatesInGPU, 5/*track candidate type pT3=5*/, pixelTripletIndex, pixelTripletIndex, &pixelTripletsInGPU.logicalLayers[5 * pixelTripletIndex], &pixelTripletsInGPU.lowerModuleIndices[5 * pixelTripletIndex], &pixelTripletsInGPU.hitIndices[10 * pixelTripletIndex], segmentsInGPU.seedIdx[pT3PixelIndex - pLS_offset], __H2F(pixelTripletsInGPU.centerX[pixelTripletIndex]), __H2F(pixelTripletsInGPU.centerY[pixelTripletIndex]),radius,trackCandidateIdx, pixelTripletIndex);
+                trackCandidateIdx -= (nPixTrackCanNotSaved > 0)*nPixTrackCanNotSaved; // Take into account pT5 candidates that were not saved
```

Comment:
This is because of what I mentioned in the previous comment. If pT5s have already exceeded `N_MAX_PIXEL_TRACK_CANDIDATES` by 10, then `trackCandidateIdx` has been increased by 10, without saving anything in those spots, so we need to decrease it to continue saving stuff in the right place.

Reply:
Ok, so I guess this becomes relevant for when we start saving the non-pixel objects? Because, I don't see where it matters for the pixel objects, if the `N_MAX_PIXEL_TRACK_CANDIDATES ` number has been exceeded.